packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -144,38 +144,30 @@ public class AmplifyStorageOperations {
         return itemAsDictionary
     }
     
-    private static func prepareError(flutterResult: @escaping FlutterResult, storageError: StorageError) {
-        let details: Dictionary<String, String> = createSerializedError(storageError: storageError);
-        logErrorContents(error: storageError)
-           ErrorUtil.postErrorToFlutterChannel(
-               result: flutterResult,
-               errorCode: "StorageException",
-               details: details)
-    }
-    
-    private static func prepareGenericError(flutterResult: @escaping FlutterResult, error: Error) {
-        logErrorContents(error: error)
+    private static func prepareError(flutterResult: @escaping FlutterResult, error: Error) {
         var message: String = ErrorMessages.missingExceptionMessage
         var recoverySuggestion: String = ErrorMessages.missingRecoverySuggestion
-        var underlyingError: Error? = nil
-
+        var details: Dictionary<String, String>
+        
+        if (error is StorageError) {
+            details = createSerializedError(storageError: error as! StorageError);
+        }
         if (error is AmplifyFlutterValidationException) {
             let e = error as! AmplifyFlutterValidationException
             message = e.errorDescription
             recoverySuggestion = e.recoverySuggestion
-            underlyingError = e.underlyingError
+            details = createSerializedError(message: message,
+                                            recoverySuggestion: recoverySuggestion,
+                                            underlyingError: "AmplifyFlutterValidationException")
```

Comment:
You should actually pass the  error and the type should also be `Error` if the param name is underlyingError, not a `String`.

Reply:
Fixed.